packages/deploy-config-sub-generator/src/prompts/sub-gen.ts

```diff
@@ -68,8 +72,15 @@ export async function getSubGenPrompts(
         options.overwrite
     );
     const indexGenerationAllowed =
-        !isLibrary && launchStandaloneFromYui && !(await indexHtmlExists(fs, options.appRootPath));
+        !isLibrary && launchStandaloneFromYui && !(await indexHtmlExists(fs, options.appRootPath)) && !isAdp;
 
+    const promptOptions: AbapDeployConfigPromptOptions = {
+        packageAutocomplete: { shouldValidatePackageForStartingPrefix: isAdp, shouldValidatePackageType: isAdp },
+        packageManual: { shouldValidatePackageForStartingPrefix: isAdp, shouldValidatePackageType: isAdp },
+        targetSystem: { shouldRestrictDifferentSystemType: isAdp },
+        transportInputChoice: { hideIfOnPremise: isAdp },
+        ui5AbapRepo: { hideIfOnPremise: isAdp }
+    };
     // ABAP prompts
     const { prompts: abapPrompts, answers: abapAnswers } = await getAbapQuestions({
```

Comment:
Could we update `getAbapQuestions` so prompt options includes `indexGenerationAllowed` `showOverwriteQuestion` and not as separate params